app/src/main/java/org/mozilla/tv/firefox/webrender/cursor/CursorView.kt

```diff
@@ -5,89 +5,103 @@
 package org.mozilla.tv.firefox.webrender.cursor
 
 import android.content.Context
-import android.os.Handler
-import android.os.Looper
-import android.os.Message
-import androidx.annotation.UiThread
-import androidx.appcompat.widget.AppCompatImageView
+import android.graphics.Canvas
+import android.graphics.PointF
 import android.util.AttributeSet
-import android.view.KeyEvent
+import androidx.annotation.CheckResult
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.core.view.isVisible
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.disposables.Disposable
+import io.reactivex.rxkotlin.Observables
+import io.reactivex.rxkotlin.addTo
 import org.mozilla.tv.firefox.R
-import org.mozilla.tv.firefox.utils.RemoteKey
-import java.lang.ref.WeakReference
 import java.util.concurrent.TimeUnit
 
-private const val HIDE_MESSAGE_ID = 0
 private const val HIDE_ANIMATION_DURATION_MILLIS = 250L
 private val HIDE_AFTER_MILLIS = TimeUnit.SECONDS.toMillis(3)
 
 /**
- * A drawn Cursor: see [LegacyCursorViewModel] for responding to keys and setting position.
- * The cursor will hide itself when it hasn't received a location update recently.
+ * Renders the cursor visible on top of the browser.
+ *
+ * Most behavior is controlled by [CursorModel]
+ *
+ * Any images set must be Bitmaps, not Drawables
  */
 class CursorView(context: Context, attrs: AttributeSet) : AppCompatImageView(context, attrs) {
 
-    private val hideHandler = CursorHideHandler(this)
+    private val position = PointF(x, y)
 
-    @UiThread
-    fun updatePosition(x: Float, y: Float) {
-        // translation* sets the top-left corner so we offset it in order to center the asset.
-        translationX = x - width / 2
-        translationY = y - height / 2
+    private var cursorModel: CursorModel? = null
+    private var width = 0f
```

Comment:
In hindsight, I think my idea to micro-optimize this is dumb: we're already calling a bunch functions in mutatePosition so it's not that helpful. Also, it forces us to duplicate an existing source of truth (which we may have gotten wrong). Also also, it makes the class less readable by adding a handful of unnecessary lines of code (var declarations, override onSizeChanged).

Reply:
It's hard to know where to draw the line on that kind of thing, but I agree that we can do without this one.